ocs-ktc/src/main/kotlin/org/ostelco/prime/ocs/core/OnlineCharging.kt

```diff
@@ -40,64 +40,84 @@ object OnlineCharging : OcsAsyncRequestConsumer {
         val msisdn = request.msisdn
 
         if (msisdn != null) {
+            if (isKeepAlive(request)) {
+                handleKeepAlive(request, returnCreditControlAnswer)
+            } else {
+                chargeRequest(request, msisdn, returnCreditControlAnswer)
+            }
+        }
+    }
 
-            val responseBuilder = CreditControlAnswerInfo.newBuilder()
 
-            responseBuilder.setRequestNumber(request.requestNumber)
+    private fun isKeepAlive(request: CreditControlRequestInfo) : Boolean {
```

Comment:
This can be just a one-line function. ;)

Reply:
fixed